website/docs/sources.md

```diff
@@ -28,6 +28,12 @@ The below example uses a [`context`](#build-context) source type.
 The root of the source is the `path/in/source` directory.
 The source will include all `.txt` files within `path/in/source` except for `secret.txt`.
 
+For each source type, the source may be treated as a directory or a file, which
+changes the handling of the source.
+The `path` field is also used to determine that.
+If `path` points at a file in the source then the source is always considered a
+file regardless of source type.
```

Comment:
Could you please add something about determining if is a directory or file is really only based on the path `field` and the `include` and `exclude` are not taken into account?

Reply:
Added an explicit callout for include and exclude.